packages/@angular/cli/commands/build.ts

```diff
@@ -235,7 +244,45 @@ const BuildCommand = Command.extend({
       ui: this.ui,
     });
 
-    return buildTask.run(commandOptions);
+
+    const buildPromise = buildTask.run(commandOptions);
+
+
+    const clientApp = getAppFromConfig(commandOptions.app);
+
+    const doAppShell = commandOptions.target === 'production' &&
+      (commandOptions.aot === undefined || commandOptions.aot === true) &&
+      !commandOptions.skipAppShell;
+    if (!clientApp.appShell || !doAppShell) {
+      return buildPromise;
+    }
+    const serverApp = getAppFromConfig(clientApp.appShell.app);
```

Comment:
Do we give a nice error message if the app doesn't exist, or if the structure of `clientApp.appShell` isn't valid?

Reply:
Not yet, but it will get added.